gsa/src/web/pages/scanners/dialog.js

```diff
@@ -140,211 +165,233 @@ class ScannerDialog extends React.Component {
-      label: scannerTypeName(scannerType),
-      value: scannerType,
-    }));
-
-    const scanner_credentials = filter_credentials(credentials, type);
-    const is_edit = isDefined(scanner);
-    const isInUse = isDefined(scanner) && scanner.isInUse();
-    const show_cred_info =
-      isDefined(scanner) &&
-      isDefined(scanner.credential) &&
-      scanner.credential.credential_type === CLIENT_CERTIFICATE_CREDENTIAL_TYPE;
-
-    const isGreenboneSensorType = type === GREENBONE_SENSOR_SCANNER_TYPE;
-    const isOspScannerType = type === OSP_SCANNER_TYPE;
-
-    if (isGreenboneSensorType) {
-      credential_id = '';
-    }
+  const scannerTypesOptions = map(SCANNER_TYPES, scannerType => ({
+    label: scannerTypeName(scannerType),
+    value: scannerType,
+  }));
+
+  const scanner_credentials = filter_credentials(credentials, type);
+  const is_edit = isDefined(scanner);
+  const isInUse = isDefined(scanner) && scanner.isInUse();
+  const show_cred_info =
+    isDefined(scanner) &&
+    isDefined(scanner.credential) &&
+    scanner.credential.credential_type === CLIENT_CERTIFICATE_CREDENTIAL_TYPE;
+
+  const isGreenboneSensorType = type === GREENBONE_SENSOR_SCANNER_TYPE;
+  const isOspScannerType = type === OSP_SCANNER_TYPE;
+
+  if (isGreenboneSensorType) {
+    credential_id = '';
+  }
+
+  const [activeTab, setActiveTab] = useState(0);
+
+  const totalPages = 2;
+
+  const {page, handlePageTurn, prevDisabled, nextDisabled} = usePageTurn({
+    activeTab,
+    lastPage: totalPages - 1,
+  });
+
+  useEffect(() => {
+    setActiveTab(page);
+  }, [page]);
+
+  return (
+    <SaveDialog
+      title={title}
+      onClose={onClose}
+      onSave={onSave}
+      multiStep={true} // the option to switch to the MultiStepFooter. Leaving it out defaults to false
+      onNext={() => handlePageTurn('Next')} // 'Next' turns the next page
+      prevDisabled={prevDisabled}
+      nextDisabled={nextDisabled}
+      onPrev={() => handlePageTurn('Previous')} // 'Previous' goes back
+      defaultValues={data}
+      values={{
+        credential_id,
+        type,
+      }}
+    >
+      {({values: state, onValueChange}) => {
+        return (
+          <Layout flex="column">
+            <DialogTabLayout grow="1" align={['start', 'end']}>
+              <TabList
+                active={activeTab}
+                align={['start', 'stretch']}
+                onActivateTab={handleActivateTab}
+              >
+                <Tab style={{backgroundColor: 'red'}}>{_('Metadata')}</Tab>
+                <Tab>{_('Scanner Type')}</Tab>
+              </TabList>
+            </DialogTabLayout>
+            <Tabs active={activeTab}>
+              <TabPanels>
+                <TabPanel>
+                  <FormGroup title={_('Name')}>
+                    <TextField
+                      name="name"
+                      grow="1"
+                      value={state.name}
+                      size="30"
```

Comment:
Could you check if setting the size here (and in the next TextField) is senseful? I don't think we are using this anymore, especially because grow="1" should render the Field in the whole available space.